### PR TITLE
[FIX] account: automatic entry wizard: avoid clashes with lock dates

### PR DESCRIPTION
**Pull request by antoine162**
Currently, if you use the automatic entry wizard to change the period of a journal item dated prior to the lock date, you'll just get blocked with a UserError, with no workaround.

This commit changes the date of the adjustment entry to be the first end of month after the lock date. As a result, the adjustment entry can be created.

Based on @poma-odoo's PR #92439 , targeting 16.0, simplifying the test and refraining from 'optimizing' the widget :smile: 
Taskid: 2823170

### CODE REVIEW

File: addons/account/tests/test_transfer_wizard.py [hunk 1]
```diff
@@ -289,3 +291,49 @@ def test_transfer_wizard_no_currency_conversion(self):
...
+
+        # Use the transfer wizard to change the entry date
+        wizard = self.env['account.automatic.entry.wizard'] \
+            .with_context(active_model='account.move.line', active_ids=move.line_ids[0].ids) \
+            .create({
+                'action': 'change_period',
+                'date': '2019-05-01',
+                'journal_id': self.company_data['default_journal_misc'].id,
+            })
+        wizard_res = wizard.do_action()
```
Comment (william-andre) [inline addons/account/tests/test_transfer_wizard.py:335]: Maybe just check that the date is displayed in the wizard, as there is no warning whatsoever anywhere else that it won't be on the date of the original move

**Review by william-andre — COMMENTED** [review]
Very nice :+1:

File: addons/account/wizard/account_automatic_entry_wizard.py [hunk 2]
```diff
@@ -284,6 +300,7 @@ def _get_move_dict_vals_change_period(self):
                 }),
             ]
 
+        reference_move.invalidate_recordset()
```
Comment (william-andre) [inline addons/account/wizard/account_automatic_entry_wizard.py:303]: Uh? What? Why?

File: addons/account/wizard/account_automatic_entry_wizard.py [hunk 3]
```diff
@@ -101,6 +102,20 @@ def _compute_account_type(self):
...
+            if record.action == 'change_period':
+                for aml in record.move_line_ids:
+                    lock_date_message = aml.move_id._get_lock_date_message(aml.date, aml.move_id._affect_tax_report())
+                    if lock_date_message:
+                        record.lock_date_message = lock_date_message
+                        break
+                else:
+                    record.lock_date_message = False
+            else:
+                record.lock_date_message = False
```
Comment (william-andre) [inline addons/account/wizard/account_automatic_entry_wizard.py:117]:
```suggestion
            if record.action == 'change_period':
                record.lock_date_message = False
                for aml in record.move_line_ids:
                    lock_date_message = aml.move_id._get_lock_date_message(aml.date, aml.move_id._affect_tax_report())
                    if lock_date_message:
                        record.lock_date_message = lock_date_message
                        break
```

File: addons/account/wizard/account_automatic_entry_wizard.py [hunk 4]
```diff
@@ -101,6 +102,20 @@ def _compute_account_type(self):
...
 
+    @api.depends('action', 'move_line_ids')
+    def _compute_lock_date_message(self):
+        for record in self:
+            if record.action == 'change_period':
+                for aml in record.move_line_ids:
+                    lock_date_message = aml.move_id._get_lock_date_message(aml.date, aml.move_id._affect_tax_report())
+                    if lock_date_message:
+                        record.lock_date_message = lock_date_message
+                        break
```
Comment (william-andre) [inline addons/account/wizard/account_automatic_entry_wizard.py:113]: The `break` is a bit weird though, we don't really know which date changed for which move

File: addons/account/wizard/account_automatic_entry_wizard.py (hunk 2 above)
Comment (william-andre) [reply addons/account/wizard/account_automatic_entry_wizard.py:303]: Yes, I'm ok for using `new()`, but why do you need to invalidate?

Comment (antoine162) [reply addons/account/wizard/account_automatic_entry_wizard.py:303]: But is the `invalidate_recordset` necessary? I don't know actually... :wink:
I'd appreciate it if you can explain hahaha

File: addons/account/wizard/account_automatic_entry_wizard.py (hunk 4 above)
Comment (antoine162) [reply addons/account/wizard/account_automatic_entry_wizard.py:113]: Yes. At least it alerts the user that something was changed. :smile:

Comment (antoine162) [reply addons/account/wizard/account_automatic_entry_wizard.py:114]: I will check with Benj what he wants in this case.

**Review by william-andre — APPROVED** [review]
@robodoo r+

Exception for the template asked to runbot team

**Comment by guva-odoo** [comment]
Hello @antoine162 shouldn't we beck port this in `15.0` as the bug you're fixing is happening too, or is there a reason not to have done it?

**Comment by antoine162** [comment]
Hey @guva-odoo it's a good idea! No, there wasn't any particular reason other than the fact that the task was for 16.0. I just asked Benj, he says it's not worth the effort back-porting this to 15, given that it's a niche use case (usually, users won't have errors in their accounting when closing a period) and users have a work-around anyway (just create the entries manually themselves)...

**Pull request closed without merging**